### PR TITLE
feat: Edit message

### PR DESCRIPTION
**Pull request by tofers**
# PR Details

Added message editing from version "2.2323.4"

## Description

Currently, message editing is only possible for text type messages. But there are other types in Whatsapp code that will be available later.

Settings are put in a separate type, we will update it as the functionality fills up.

Instructions Install the branch:
```
NPM: npm i github:tofers/whatsapp-web.js#editMsg
Yarn: yarn add github:tofers/whatsapp-web.js#editMsg
```
Message change event
```
client.on('message_edit', async (msg, newBody, prevBody) => {
    console.log('message', msg );
    console.log('message text', prevBody, newBody);
});
```

Example usage
```
client.on('message', async msg => {
    console.log('MESSAGE RECEIVED', msg);
    if (msg.body === '!edit') {
        if (msg.hasQuotedMsg) {
            const quotedMsg = await msg.getQuotedMessage();
            if (quotedMsg.fromMe) {
                quotedMsg.edit(msg.body.replace('!edit', ''))
            } else {
                msg.reply('I can only delete my own messages');
            }
        }
    }
});
```

## Related question

Haven't seen anyone ask that yet. But I need this functionality.

## Motivation and Context

To be able to update a message 

## How has this been tested

I tested it on linux systems

## Types of changes

- [ ] Change dependencies
- [ ] Bug fix (non-breaking change that fixes the problem)
- [x] New feature (non-breaching change that adds functionality)
- [x] Violating change (a fix or feature that will change existing functionality)

## Checklist

- [x] My code matches the code style of this project.
- [x] I have updated the documentation accordingly (index.d.ts).

### CODE REVIEW

**Comment by seanesbo** [comment]
Good job, works perfect!!

**Comment by rapkido** [comment]
Tested and works great! Ty.

**Comment by Mohammed-800** [comment]
I successful check

File: src/util/Injected.js [hunk 1]
```diff
@@ -119,9 +119,9 @@ exports.LoadUtils = () => {
                     forceDocument: options.sendMediaAsDocument,
                     forceGif: options.sendVideoAsGif
                 });
-
```
Comment (tuyuribr) [inline src/util/Injected.js:122]: revert the whitespaces please

**Comment by tuyuribr** [comment]
I runned some tests, and It seems whatsapp is rate limiting the frequency which you can edit a message, should we place checkers?

**Comment by tofers** [comment]
> I runned some tests, and It seems whatsapp is rate limiting the frequency which you can edit a message, should we place checkers?

I'll take a look. But we are not restricting the invocation of the other methods. I don't think we need any extra check on the interval between edits.

**Comment by tofers** [comment]
[#2275](https://github.com/pedroslopez/whatsapp-web.js/issues/2275)

**Review by tuyuribr — APPROVED** [review]
LGTM

File: example.js [hunk 2]
```diff
@@ -201,6 +201,15 @@ client.on('message', async msg => {
         client.sendMessage(msg.from, list);
     } else if (msg.body === '!reaction') {
         msg.react('👍');
+    } else if (msg.body === '!edit') {
+        if (msg.hasQuotedMsg) {
+            const quotedMsg = await msg.getQuotedMessage();
+            if (quotedMsg.fromMe) {
+                quotedMsg.edit(msg.body.replace('!edit', ''));
+            } else {
+                msg.reply('I can only delete my own messages');
```
Comment (tuyuribr) [inline example.js:210]: I can only edit my own messages

**Review by CSFelix — COMMENTED** [review]
LGTM

**Review by aliyss — APPROVED** [review]
LGTM

**Comment by alamia442** [comment]
edit message only work for quotedMsg?

how to edit previous sending message by client.on('message_create'?

**Pull request merged**